4.2.5/docker-compose.yml

```diff
@@ -0,0 +1,159 @@
+version: '3.1'
+
+volumes:
+  geonetwork:
+  esdata:
+  pgdata:
+  pglog:
+  nginxlog:
+  apachelog:
+
+
+networks:
+  gn-network:
+    driver: bridge
+
+
+services:
+  www:
+    image: nginx
+    ports:
+      - 80:80
+    environment:
+      # Only used for / redirect to default webapp
+      - NGINX_PUBLIC_HOST=195.201.225.214
+      #- NGINX_PUBLIC_HOST=apps.titellus.net
+      - NGINX_HOST=195.201.225.214
+      - NGINX_PORT=80
+    volumes:
+      - ./nginx:/etc/nginx/templates
+      - nginxlog:/var/log/nginx
+    depends_on:
+      - geonetwork
+    networks:
+      - gn-network
+
+#  www:
+#    image: httpd
+#    ports:
+#      - 81:80
+#    volumes:
+#      - ./apache/httpd.conf:/usr/local/apache2/conf/httpd.conf
+#      - ./apache/gn-httpd.conf:/usr/local/apache2/conf/gn-httpd.conf
+#      - apachelog:/usr/local/apache2/logs
+#    depends_on:
+#      - geonetwork
+#    networks:
+#      - gn-network
+
+
+  database:
+    image: postgis/postgis:11-2.5
+    environment:
+      POSTGRES_USER: geonetwork
+      POSTGRES_PASSWORD: geonetwork
+      POSTGRES_DB: geonetwork
+    command: [ "postgres",
+               "-c", "log_statement=all",
+               "-c", "logging_collector=true",
+               "-c", "log_file_mode=0644",
+               "-c", "log_directory=/var/log/postgresql",
+               "-c", "log_filename=postgresql.log" ]
+    volumes:
+      - pgdata:/var/lib/postgresql/data
+      - pglog:/var/log/postgresql
+    networks:
+      - gn-network
+
+
+  geonetwork:
+    image: geonetwork:4.2.2
+    restart: always
+    ports:
+      - 8080:8080
+    environment:
+      DATA_DIR: /catalogue-data
+
+      JAVA_OPTS: >
```

Comment:
Should we use JVM args here ?

```
      JAVA_OPTS: >
        -Dorg.eclipse.jetty.annotations.AnnotationParser.LEVEL=OFF
        -Djava.security.egd=file:/dev/./urandom -Djava.awt.headless=true
        -Xms512M -Xss512M -Xmx2G -XX:+UseConcMarkSweepGC
        -Dgeonetwork.dir=/catalogue-data
        -Dgeonetwork.formatter.dir=/catalogue-data/data/formatter
        -Dgeonetwork.codeList.dir=/opt/geonetwork/WEB-INF/data/config/schema_plugins
        -Dgeonetwork.indexConfig.dir=/opt/geonetwork/WEB-INF/data/config/index
        -Des.protocol=http 
        -Des.port=9200 
        -Des.url=http://elasticsearch:9200 
        -Dgeonetwork.ESFeaturesProxy.targetUri=http://elasticsearch:9200/geo-features/{_}
        -Dgeonetwork.HttpDashboardProxy.targetUri=http://kibana:5601
        -Dgeonetwork.MicroServicesProxy.targetUri=http://ogc-api-records-service:8080

      GEONETWORK_DB_TYPE: postgres
      GEONETWORK_DB_HOST: database
      GEONETWORK_DB_PORT: 5432
      GEONETWORK_DB_NAME: geonetwork
      GEONETWORK_DB_USERNAME: geonetwork
      GEONETWORK_DB_PASSWORD: geonetwork

      WEBAPP_CONTEXT_PATH: geonetwork
```

Reply:
We can use it, but I'd leave it to the next big release, 4.4.x or 5.0.x, just to avoid changing the users config in a minor release.